mysociety/caps
# Prepare for new 2023 Councils

This PR imports new data (and some new data structures) to cope with the council structure changes in April 2023 (https://github.com/mysociety/caps/issues/492).

- Emissions and similarity datasets have been updated. 
- CAPE given awareness that some councils are no longer 'current'.
- New language to link to predecessor and successor councils.

Notes:
- Needs another pass to sort out those tests.
- I've fixed up the tests for the new 'start-date' column, and to remove the bit that says councils will be removed - but don't quite understand why this has broken the 'number of councils' affected tests. @struan ?

To test:

Run: ```script/update --all``` to improve the new emissions/similarity data. 

To merge:

Sometime after merging in April, the supporting datasets should have the prerelease datasets properly released - and then references here should also be updated. 

## Screenshots

Shows new council, not current county council:

![image](https://user-images.githubusercontent.com/8157058/223778771-ea44e99b-fbc1-4399-a050-19bfe1e9afca.png)

Links to former council pages:

![image](https://user-images.githubusercontent.com/8157058/223778944-febe1f1e-6d4f-4689-b6b5-1897d7575872.png)

Links to successor council pages:

![image](https://user-images.githubusercontent.com/8157058/223779043-527a2190-a0ad-4307-8045-683f0ef40e54.png)

Postcode goes just to new councils (even when two overlapped before, if both are now gone):

![image](https://user-images.githubusercontent.com/8157058/223780267-a86b22cb-22f8-42bc-8449-993cbd6b2702.png)

![image](https://user-images.githubusercontent.com/8157058/223780306-0ca07836-3936-4b1e-9ceb-c5be20cdda3c.png)

New emissions data:

![image](https://user-images.githubusercontent.com/8157058/223780768-279a52f7-2f2f-4a32-9dda-4f8470e3f0cd.png)

New similarity data:

![image](https://user-images.githubusercontent.com/8157058/223780993-c2553d77-2879-49d6-bcc3-c5a849559f96.png)
